RecoTracker/MkFitCore/src/MkBuilder.cc

```diff
@@ -800,7 +799,9 @@ namespace mkfit {
                                                          layer_plan_it.is_pickup_only(),
                                                          iteration_dir);
 
-          if (layer_plan_it.is_pickup_only() || theEndCand == 0)
+          dprintf("  Number of candidates to process: %d, nHits in layer: %d\n", theEndCand, layer_of_hits.nHits());
+
+          if (layer_plan_it.is_pickup_only() || theEndCand == 0 || layer_of_hits.nHits() == 0)
```

Comment:
is the skip here on empty layer completely benign? Would we appropriately get the invalid/missing hits and material updates processed?